test/TestUtils.ts

```diff
@@ -0,0 +1,394 @@
+
+const startRelay = async (relayHub: RelayHub, options: StartRelayParams) => {
+  const { serverConfig, delay, stake, relayOwner } = options;
+
+  fs.rmSync(SERVER_WORK_DIR, { recursive: true, force: true });
+
+  const originalConfig = { ...config };
+
+  config.util.extendDeep(originalConfig, {
+    ...serverConfig,
+    contracts: {
+      relayHubAddress: relayHub.address,
+    },
+    app: {
+      workdir: SERVER_WORK_DIR,
+      devMode: true,
+      checkInterval: 10,
+      logLevel: 5,
+    },
+  });
+
+  const url = config.get<string>('app.url');
+
+  const runServerPath = path.resolve(
+    __dirname,
+    '../node_modules/@rsksmart/rif-relay-server/dist/commands/Start.js'
+  );
+
+  const proc: ChildProcessWithoutNullStreams & { alreadyStarted?: number } =
+    childProcess.spawn('node', [runServerPath]);
+
+  const { relayManagerAddress } = await verifyRelayServerStatus(url, 3);
+
+  console.log('Relay Server Manager Address', relayManagerAddress);
+
+  await provider.getSigner(relayOwner).sendTransaction({
+    to: relayManagerAddress,
+    value: utils.parseEther('2'),
+  });
+
+  await relayHub.stakeForAddress(relayManagerAddress, delay || 2000, {
+    from: relayOwner,
+    value: stake || utils.parseEther('1'),
+  });
+
+  const { ready, relayWorkerAddress } = await verifyRelayServerStatus(
+    url,
+    25,
+    500,
+    true
+  );
+
+  expect(ready, 'Timed out waiting for relay to get staked and registered').is
+    .ok;
+
+  return {
+    proc,
+    worker: relayWorkerAddress,
+    manager: relayManagerAddress,
+  };
+};
+
+const stopRelay = (proc: ChildProcessWithoutNullStreams): void => {
+  proc.kill();
+};
+
+const evmMine = async (): Promise<unknown> => {
+  return provider.send('evm_mine', []);
+};
+
+const evmMineMany = async (count: number): Promise<void> => {
+  for (let i = 0; i < count; i++) {
+    await evmMine();
+  }
+};
+
+const increaseBlockchainTime = async (time: number): Promise<void> => {
+  await provider.send('evm_increaseTime', [time]);
+  await evmMine();
+};
+
+const createSnapshot = async (): Promise<string> => {
+  return (await provider.send('evm_snapshot', [])) as string;
+};
+
+const revertSnapshot = async (snapshotId: string) => {
+  return (await provider.send('evm_revert', [snapshotId])) as boolean;
+};
+
+const verifyRelayServerStatus = async (
+  url: string,
+  attempts: number,
+  interval = 1000,
+  isReady = false
+): Promise<HubInfo> => {
+  let response: HubInfo | undefined;
+  const httpClient = new HttpClient(new HttpWrapper(undefined, 'silent'));
+
+  for (let i = 0; i < attempts; i++) {
+    try {
+      console.log('sleep before cont.');
+      await sleep(interval);
+      response = await httpClient.getChainInfo(url);
+
+      if (isReady && !response.ready) {
+        continue;
+      }
+
+      return response;
+    } catch (e) {
+      /* empty */
+    }
+  }
+
+  expect(response, "can't ping server").to.be.ok;
+
+  throw Error("can't ping server");
```

Comment:
I think we could either expect something or throw the error. I think that using both may be redundant

Reply:
I would prefer to throw error, I think that I have this approach on multiple places, Ill update it.